packages/client/screens/MusicPlaylistEditorRoomsSearchScreen/index.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState } from 'react';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { Text, View } from 'dripsy';
+import { FlatList, TouchableOpacity } from 'react-native';
+import { useActor, useMachine } from '@xstate/react';
+import { ActorRef } from 'xstate';
+import { MpeRoomSummary } from '@musicroom/types';
+import { AppScreenWithSearchBar } from '../../components/kit';
+import { MpeTabMpeRoomsScreenProps } from '../../types';
+import { useMusicPlaylistsActor } from '../../hooks/useMusicPlaylistsActor';
+import {
+    AppScreenHeaderWithSearchBarMachineEvent,
+    AppScreenHeaderWithSearchBarMachineState,
+} from '../../machines/appScreenHeaderWithSearchBarMachine';
+import { mpeRoomSearchMachine } from '../../machines/mpeRoomUniversalSearchMachine';
+
+interface PlaylistListItemProps {
+    roomSummary: MpeRoomSummary;
+    onPress: (roomSummary: MpeRoomSummary) => void;
+}
+
+const PlaylistListItem: React.FC<PlaylistListItemProps> = ({
+    roomSummary,
+    onPress,
+}) => {
+    const { roomID, roomName } = roomSummary;
+    return (
+        <TouchableOpacity
+            testID={`mpe-room-card-${roomID}`}
+            onPress={() => {
+                onPress(roomSummary);
+            }}
+        >
+            <View>
+                <Text sx={{ color: 'white' }}>{roomName}</Text>
+            </View>
+        </TouchableOpacity>
+    );
+};
+
+const MusicPlaylistEditorRoomsSearchScreen: React.FC<MpeTabMpeRoomsScreenProps> =
+    ({ navigation }) => {
+        const insets = useSafeAreaInsets();
+        const [screenOffsetY, setScreenOffsetY] = useState(0);
+
+        //Library Search machine
+        const [libraryRoomState, libraryRoomsSend] =
+            useMachine(mpeRoomSearchMachine);
+        const hasMoreRoomsToFetch = false;
+        const searchBarActor: ActorRef<
+            AppScreenHeaderWithSearchBarMachineEvent,
+            AppScreenHeaderWithSearchBarMachineState
+        > = libraryRoomState.children.searchBarMachine;
+        const [searchState, sendToSearch] = useActor(searchBarActor);
+        const showHeader = searchState.hasTag('showHeaderTitle');
+        ///
+
+        const { appMusicPlaylistsActorRef } = useMusicPlaylistsActor();
+        function handleRoomPress({ roomID, roomName }: MpeRoomSummary) {
+            appMusicPlaylistsActorRef.send({
+                type: 'DISPLAY_MPE_ROOM_VIEW',
+                roomID,
+                roomName,
+            });
+        }
+
+        useEffect(() => {
+            libraryRoomsSend({
+                type: 'FETCH_ROOMS',
+            });
+        }, [libraryRoomsSend]);
```

Comment:
The purpose of this `useEffect` is to fetch rooms each time we go back to the screen?

Reply:
Yes

Reply:
Okay!
I think the `useEffect` is not necessary because the initial state of the machine is `fetchingRooms` and the state machine will be interpreted by `useMachine` each time the screen is created, and will be stopped when the component is destroyed. Those are the lines in `useInterpret`, used inside `useMachine`, that deal with starting and stopping the service: <https://github.com/statelyai/xstate/blob/89fcc6d47f1890f09752d73d086fdd158898c194/packages/xstate-react/src/useInterpret.ts#L101-L120>.

That's not a big deal, I just wanted to understand this part of the screen code.

Reply:
I'm not sure this will be the case here, as we're exporting const machines instances
I've done this as I've been struggling on this point, if I only export a createMachine functionn between each rerender the value passed to useMachine will be different and xState will raise warnings

```ts
export const libraryMpeRoomSearchMachine = createMpeRoomUniversalSearchMachine({
    fetchMpeRooms: fetchLibraryMpeRooms,
});

export const mpeRoomSearchMachine = createMpeRoomUniversalSearchMachine({
    fetchMpeRooms: fetchAllMpeRooms,
});

```

Reply:
Indeed, state machines _definitions_ need to be consistent between renders.
But state machines we provide to `useMachine` or `useIntepret` hooks are only definitions. They do nothing by themselves, they need to be _interpreted_.
This is what `useMachine` and `useInterpret` do, they interpret the machine definition into a _service_ and keep it alive for all the life of the component they are called in.

Here is a demonstration without the `useEffect`: <https://cln.sh/sNh8n7>

Reply:
Indeed you're right !
I've been struggling on displaying the mpe-room-card results in tests, I assume I've fixed the problem in // and kept in mind this was this specific usage I've been setting up that was solving the issue
Thanks, about to update on that point